Style engine: updating and fleshing out docs (#44405)
WordPress/gutenberg: packages/style-engine/docs/using-the-style-engine-with-block-supports.md

*/
```

## Use case
When [registering a block support](https://developer.wordpress.org/reference/classes/wp_block_supports/register/), it is possible to pass an 'apply' callback in the block support config array to add or extend block support attributes with "class" or "style" properties.

If a block has opted into the block support, the values of "class" and "style" will be applied to the block element's "class" and "style" attributes accordingly when rendered in the frontend HTML. Note, this applies only to server-side rendered blocks, for example, the [Site Title block](https://developer.wordpress.org/block-editor/reference-guides/core-blocks/#site-title).

> The callback receives `$block_attributes` as an argument. `$block_attributes` only contains the styles object, if any, and not any CSS or classnames to be applied to the block's HTML elements. 

Reviewer: "`$block_attributes` only contains the styles object" reads slightly ambiguous to me, as the array contains not just the styles object but also any other block attributes. Would something like the following work?

```suggestion
The callback receives `$block_type` and `$block_attributes` as arguments. The `style` attribute within `$block_attributes` only contains the raw style object, if any styles have been set for the block, and not any CSS or classnames to be applied to the block's HTML elements. 
```

> Here is where `wp_style_engine_get_styles` comes in handy: it will generate CSS and, if appropriate, classnames to be added to the "style" and "class" HTML attributes.

Reviewer: `style` can potentially be confusing (between block attributes and rendered block markup), so a totally optional suggestion to see if we can clarify it a bit more here:

```suggestion
Here is where `wp_style_engine_get_styles` comes in handy: it will generate CSS and, if appropriate, classnames to be added to the "style" and "class" HTML attributes in the final rendered block markup.
```

> Here is a _very_ simplified version of how [color block supports](https://github.com/WordPress/gutenberg/tree/HEAD/lib/block-supports/color.php) works:

Reviewer: Tiny nit: plural versus singular

```suggestion
Here is a _very_ simplified version of how the [color block support](https://github.com/WordPress/gutenberg/tree/HEAD/lib/block-supports/color.php) works:
```

Author: My brain also did a double take but the third person singular was intentional, as I was referring to the [block supports API](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-supports/).

But yeah without knowing about the block supports API as an entity it sounds a bit daft. Thanks for the suggestion!
